pkg/apis/core/validation/validation.go

```diff
@@ -4294,6 +4302,73 @@ func ValidatePodSpec(spec *core.PodSpec, podMeta *metav1.ObjectMeta, fldPath *fi
 	return allErrs
 }
 
+func validatePodResources(spec *core.PodSpec, podClaimNames sets.Set[string], fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
+	if spec.Resources == nil {
+		return nil
+	}
+
```

Comment:
I think you also want to ensure that `claims` is not set (`field.Forbidden()`) ?